boards/hifive1/src/main.rs

```diff
@@ -118,6 +120,56 @@ impl KernelResources<e310_g002::chip::E310x<'static, E310G002DefaultPeripherals<
     }
 }
 
+/// This is in a separate, inline(never) function so that its stack frame is
+/// removed when this function returns. Otherwise, the stack space used for
+/// these static_inits is wasted.
+#[inline(never)]
+unsafe fn get_peripherals() -> &'static mut E310G002DefaultPeripherals<'static> {
+    static_init!(
+        E310G002DefaultPeripherals,
+        E310G002DefaultPeripherals::new()
+    )
+}
+
+/// For the HiFive1, if load_process is inlined, it leads to really large stack utilization in
+/// main. By wrapping it in a non-inlined function, this reduces the stack utilization once
+/// processes are running.
+#[inline(never)]
+unsafe fn load_processes_not_inlined<C: Chip>(board_kernel: &'static Kernel, chip: &'static C) {
```

Comment:
I would not have this as an unsafe fn, as there is no scenario in which this is unsafe to call. Instead just wrap the slice constructions in unsafe blocks

Reply:
I've updated the code. I've moved the slices out of the `load_processes` call to reduce the clutter of keywords in `load_processes` and wrapped them in `unsafe {}`, and I wrapped the `PROCESSES` reference bit with `unsafe {}` inline as that was a shorter line. This allowed me to mark the `load_processes_not_inlined` function as safe.

Stack size and final binary size are unmodified.